google/cloud/storage/tests/curl_download_request_integration_test.cc

```diff
@@ -72,6 +74,65 @@ TEST(CurlDownloadRequestTest, SimpleStream) {
   EXPECT_EQ(kDownloadedLines, count);
 }
 
+// Run one attempt of the Regression7501 test. This is wrapped in a retry loop,
+// as integration tests flake due to unrelated (and unavoidable) problems, e.g.,
+// trying to setup connections.
+Status AttemptRegression7501() {
+  // Download the maximum number of lines supported by httpbin.org
+  auto constexpr kDownloadedLines = 100;
+  auto constexpr kTestPoolSize = 32;
```

Comment:
So this test is making the assumption that the same handle is used across multiple download requests. If so, would you like to assert that?

Reply:
Done.  But I would have been happy to just test "the factory returns a usable handle".  If the handle is usable because it is brand new, or because it was completely cleaned up then 🤷 .